src/hooks/useLocalSearch.ts

```diff
@@ -0,0 +1,22 @@
+import React from "react";
+
+const parseQuery = (query: string) =>
+  query.replace(/([.?*+\-=:^$\\[\]<>(){}|])/g, "\\$&");
+
+export function useLocalSearch<T>(
```

Comment:
I think we should start making some descriptions for generic utility functions/hooks :)

```suggestion
/**
 * Find element in array
 */
export function useLocalSearch<T>(
```